code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -0,0 +1,370 @@
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/random_shots()
+	for(var/turf/turf in range(12,get_turf(src)))
+		if(prob(5))
+			shoot_projectile(turf)
+
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/blast()
+	spawn()
+		for(var/turf/turf in range(1, target))
+			shoot_projectile(turf)
+
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/diagonals()
+	var/turf/T = locate(x + 2, y + 2, z)
+	shoot_projectile(T)
+	T = locate(x + 2, y  -2, z)
+	shoot_projectile(T)
+	T = locate(x - 2, y + 2, z)
+	shoot_projectile(T)
+	T = locate(x - 2, y - 2, z)
+	shoot_projectile(T)
+
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/cardinals()
+	var/list/attack_dirs = list(NORTH,EAST,SOUTH,WEST)
+	for(var/d in attack_dirs)
+		var/turf/E = get_edge_target_turf(src, d)
+		shoot_projectile(E)
+
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/telegraph()
+	for(var/mob/M in range(10,src))
+		if(M.client)
+			M.client.color = rgb(200, 0, 0)
+			spawn(1)
+				M.client.color = initial(M.client.color)
+			shake_camera(M, 4, 3)
+	playsound(get_turf(src),'sound/magic/clockwork/narsie_attack.ogg', 200, 1)
+
+
+/mob/living/simple_animal/hostile/megafauna/colossus/proc/seeking_laser()
+	visible_message("<span class='cult'><font size=5>\"<b>You can't run.</b>\"</font></span>")
+	var/obj/effect/overlay/temp/seeking_beam/B = new(get_turf(src))
+	B.target = target
+	B.firer = src
+
+/obj/effect/overlay/temp/seeking_beam
+	name = "disintegration ball"
+	desc = "Get out of the way!"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "shield-cult"
+	density = 1
+	luminosity = 1
+	var/target
+	var/firer
+	duration = 100
+
+/obj/effect/overlay/temp/seeking_beam/New()
+	..()
+	spawn()
+		while(loc)
+			step_towards(src, target)
+			sleep(1)
+
+/obj/effect/overlay/temp/seeking_beam/process()
+	step_towards(src, target)
+
+/obj/effect/overlay/temp/seeking_beam/Crossed(AM as mob|obj)
+	..()
+	collision(AM)
+
+/obj/effect/overlay/temp/seeking_beam/Bumped(AM as mob|obj)
+	..()
+	collision(AM)
+
+/obj/effect/overlay/temp/seeking_beam/Bump(AM as mob|obj)
+	..()
+	collision(AM)
+
+/obj/effect/overlay/temp/seeking_beam/proc/collision(AM as mob|obj)
+	if(AM == firer)
+		return
+	if(isliving(AM))
+		var/mob/living/M = AM
+		visible_message("<span class='danger'>[M] is disintegrated by the disintegration ball!</span>")
+		M.dust()
+	else if(istype(AM, /obj/item))
+		var/obj/item/I = AM
+		I.burn()
+
+/obj/item/projectile/colossus
+	name ="death bolt"
+	icon_state= "chronobolt"
+	damage = 25
+	armour_penetration = 100
+	speed = 2
+	eyeblur = 0
+	damage_type = BRUTE
+	pass_flags = PASSTABLE
+
+/obj/item/projectile/colossus/on_hit(atom/target, blocked = 0)
+	. = ..()
+	if(istype(target,/turf/)||istype(target,/obj/structure/))
+		target.ex_act(2)
+
+
+/obj/item/device/gps/internal/colossus
+	icon_state = null
+	gpstag = "Angelic Signal"
+	desc = "Get in the fucking robot."
+	invisibility = 100
+
+
+
+//Black Box
+
+/obj/machinery/smartfridge/black_box
```

Comment:
Accepting/dispensing items and displaying them in a menu is all reused from the fridge though